### PR TITLE
Improve KEB chart install test

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

The KEB chart install test workflow is currently triggered after the images are built in the same workflow that uses `pull_request_target`. Good practice is to avoid using `pull_request_target` when not needed. This PR allows to start the KEB chart install test in parallel with the image builder workflow. It waits for the images to be ready. It also makes the workflow easier to maintain when it's not locked behind `pull_request_target`.

Changes proposed in this pull request:

- Move the KEB chart install tests to a new workflow triggered by `pull_request`
- Add a script to the KEB chart install tests that checks the status of the image builder and waits for the images to be ready

**Related issue(s)**

### CODE REVIEW

File: scripts/wait_for_images.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+REPOSITORY=${REPOSITORY:-kyma-project/kyma-environment-broker}
+HEAD_SHA=$(gh pr view $PR_NUMBER --repo $REPOSITORY --json headRefOid --jq '.headRefOid')
+
+if [ -z "$HEAD_SHA" ]; then
+  echo "Failed to get the head SHA of the pull request"
+  exit 1
+fi
+
+echo "Sleeping for 3 minutes to wait for the image build before checking the status"
+sleep 180
```
Comment (MarekMichali) [inline scripts/wait_for_images.sh:20]: Image builder always runs longer than 3 minutes, so we don't want to call the API before that time.

File: scripts/wait_for_images.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+while true; do
+  WORKFLOW_RUN=$(gh run list --repo $REPOSITORY --json name,status,conclusion,createdAt,headSha --jq '[.[] | select(.name == "pull-build-images" and .headSha == "'"$HEAD_SHA"'")] | sort_by(.createdAt) | last | {name: .name, status: .status, conclusion: .conclusion, created_at: .createdAt}')
```
Comment (MarekMichali) [inline scripts/wait_for_images.sh:22]: The name is `pull-build-images`, so after the PR is merged the test will work.

File: docs/contributor/04-10-workflows.md [hunk 3]
```diff
@@ -66,9 +66,13 @@ This [workflow](/.github/workflows/run-govulncheck.yaml) runs the Govulncheck. I
...
 
-## Image Build and Test Workflow
+## Image Build Workflow
 
-This [workflow](/.github/workflows/run-govulncheck.yaml) builds images and calls the reusable [workflow](/.github/workflows/run-keb-chart-install-tests-reusable.yaml) to install the KEB chart with the new images in the k3s cluster. 
+This [workflow](/.github/workflows/pull-build-images.yaml) builds images.
+
+## KEB Chart Install Test
+
+This [workflow](/.github/workflows/run-keb-chart-install-tests.yaml) calls the reusable [workflow](/.github/workflows/run-keb-chart-install-tests-reusable.yaml) to install the KEB chart with the new images in the k3s cluster.
```
Comment (IwonaLanger) [inline docs/contributor/04-10-workflows.md:75]:
```suggestion
This [workflow](/.github/workflows/run-keb-chart-install-tests.yaml) calls the [reusable workflow](/.github/workflows/run-keb-chart-install-tests-reusable.yaml) to install the KEB chart with the new images in the k3s cluster. 
```

**Review by IwonaLanger — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**